### ISSUE TITLE
drum synths

### ISSUE THREAD

**Issue by Reaper10** [hydrogen-music/hydrogen#1990]
This is Just an idea. I'm all for this being closed if the actual programmers don't think this is a good idea. What if you took the synthedrum drum apart from the c? MT library from To make a half way decent syndrom Engine, there might be even more open source in sympathy plug in.You might be able to look at

CMT 1223	syndrum

**Comment by theGreatWhiteShark** [comment]
Hey @Reaper10 ,

As stated in #1985 we have to do quite a number of UI/UX changes alongside the introduction of synth instruments. Just having a new feature hidden in some submenu is neither very useful for most users nor the direction we are heading with the Hydrogen project.

That said, I really like the idea and it's splendid timing for you to bring it up. I'll do a redesign of some parts of instrument and pattern editor for the next major version and now that I'm aware of it I can account for this (maybe) future feature.

I actually worked on integrating CMT plugins at one point (#1453) but ultimately decided against it. I think we need a general support for plugins (including VST, LV2 etc.) first. Shipping some integrated (low end) ones is nice. But in the end this is just an appetizer leaving most users craving for more and disappointed.

**Comment by Reaper10** [comment]
That’s just thinking about this for kind of way that once the hydrogen drum machine has the basic physical modeling engine in it to there’s a button or some part of the u. I the automatically change is over or records. The output of a physical modeling engine over to the sampling.  Engine that you guys have already have that way. You can save the drum kit. As a simple hydrogen drumket, that way, if you’re using another piece of software that has support for the basic sampling engine but maybe that’s going to have the exact same engine that puts in a layer of compatibility.

**Comment by Reaper10** [comment]
I just thought of this idea. So, I’m just gonna put this as a comment on the main thread of this issue. What if you guys decided to port the LMMS kicker synthesizer, after all, it’s a percussion focus synthesizer might be an easy first try to make a synthesizer. Engine. Because the basic engines have already been built. You don’t have to start from scratch. That’s just an idea. Someone that actually does the coating for hydrogen. If you would tell me if this is a good idea or this is impossible. I’ll even give you if I can find it. [The source code](https://github.com/LMMS/lmms/tree/master/plugins/Kicker) for the LMMS kicker synth you even presets

**Comment by Reaper10** [comment]
Years ago I had designed a whole drum machine sound using nothing but l m m s’s triple oscillator synthesizer, so I think it might be a good idea to see if you can fort the triple oscillator am I still use l m m s so I might look through some of their other. Synthesizers, and see if I can come up with some good drum machine. Sound patches
If you think this is a good idea, I’ve even found the source code for the triple. Oscillator synthesizer for you
https://github.com/LMMS/lmms/blob/master/plugins/TripleOscillator/TripleOscillator.cpp

**Comment by Reaper10** [comment]
[lmms drums.zip](https://github.com/user-attachments/files/31354933/lmms.drums.zip)

**Comment by theGreatWhiteShark** [comment]
Hey @Reaper10,

at a very early state Hydrogen did had a synth (at least there was a corresponding source file). But it was removed long ago.

Anyway, current development did actually led into another direction. I currently work on making Hydrogen itself a CLAP, LV2, and VST3 plugin. The prototype is far from done but is already looking promising. In case this feature does land - and that's quite likely - the plugin support within Hydrogen (LADSPA) will be deprecated in the 1.2 release line and removed in 2.0. When Hydrogen can live within a DAW, we do not need to fiddle with plugins and synths ourselves but can focus on sequencing and sampling.

**Comment by Reaper10** [comment]
> Hey [@Reaper10](https://github.com/Reaper10),
> 
> at a very early state Hydrogen did had a synth (at least there was a corresponding source file). But it was removed long ago.
> 
> Anyway, current development did actually led into another direction. I currently work on making Hydrogen itself a CLAP, LV2, and VST3 plugin. The prototype is far from done but is already looking promising. In case this feature does land - and that's quite likely - the plugin support within Hydrogen (LADSPA) will be deprecated in the 1.2 release line and removed in 2.0. When Hydrogen can live within a DAW, we do not need to fiddle with plugins and synths ourselves but can focus on sequencing and sampling.

Okay, should I just close? All my feature requests for other types of synth engines programming language, or is that something you guys wanna look into? Just to add that much more power #2217 #1985

**Comment by theGreatWhiteShark** [comment]
> Okay, should I just close?

No, I'll do it once the feature does actually land. Until them they are still valid

**Comment by Reaper10** [comment]
> > Okay, should I just close?
> 
> No, I'll do it once the feature does actually land. Until them they are still valid

It’s not okay. I still maintain, I think, even after it becomes a little plug in that you should look into doing. Hey synthetin, or a spectrally synthed engine or some combination of both even for the bug inversion, or even adding the ability to do physical modeling, based the only reason I think these would be an improvement over the basic. Sampling, because there’s some percussion instruments that samples are not exactly easy to come by, or sometimes sometimes don’t have the equipment to sample something, so it’d be a nice, easy way to make a sample or create a syntin in the case of the physical modeling engine. But if you don’t want to do that, once it becomes a plug in, I understand I just think it’d be good to have as many possible options. Even as a plug in as possibleSome doctor mentioned the idea of a 3D stage page. I think that’d be a great thing because you could make a drum set. Sound like it’s coming from a certain type of area that physically cannot set up a real drum set in
